src/server/pps/cmds/cmds.go

```diff
@@ -881,6 +881,27 @@ All jobs created by a pipeline will create commits in the pipeline's output repo
 				fmt.Println("Pipeline unchanged, no update will be performed.")
 				return nil
 			}
+			// May not change the project name, but if it is omitted
+			// then it is considered unchanged.
+			project := request.Pipeline.GetProject()
+			projectName := project.GetName()
+			if projectName == "" {
+				if project == nil {
+					project = new(pfs.Project)
+				}
+				project.Name = pipelineInfo.Pipeline.GetProject().GetName()
+				request.Pipeline.Project = project
+			} else if projectName != pipelineInfo.Pipeline.GetProject().GetName() {
```

Comment:
Can we move the exceptional cases up?
```Go
if projectName != pipelineInfo.Pipeline.GetProject().GetName() {
    return errors.New("may not change project name")
}
if pipelineName != pipelineInfo.Pipeline.GetName() {
     return errors.New("may not change pipeline name")
}

// business logic ...
```

Reply:
Also given that we don't allow the pipeline/project name to change, I think we can safely assign it to the request without conditionals.

Reply:
+1